File: 01 MATLAB/01 Data Procurement/Yahoo_Data.m
% Extracting Yahoo Data
% Michael Beven
% 20151016

clear;
close all;

%% Define Parameters

% Start and End Dates
startDate = '2015-08-1';
endDate = '2015-08-31';

% extract data
symbol = 'IBM';
[raw_data] = fetch(yahoo,symbol,startDate,endDate);
raw_data2 = double(raw_data);

% create headers
headers = {'Open','High','Low','Close','Volume', 'Adj_Close'};

% turn to financial time series
% tsdata = fints(raw_data(:,1),raw_data(:,2:end),headers);

% convert serial date number to date
date_vector = datestr(raw_data(:,1),'yyyy-mm-dd');


% attach headers to data
% data = dataset({raw_data,headers{:}});

File: 02 R/02 Data Extraction/Yahoo_Data.R
# Querying YAHOO! Finance for data
# Michael Beven
# 20151017

Yahoo_Data = function(start_date,end_date,symbol){

# Parameters
URL = paste("http://ichart.finance.yahoo.com/table.csv?s=",symbol,sep="")

# Read in data
data.info = read.csv(URL)
data.info$Date = as.Date(data.info$Date, "%Y-%m-%d")

# Get subset of data
sub.data.info = subset(data.info, data.info$Date >= as.Date(start_date) & 
                         data.info$Date <= as.Date(end_date))


return(sub.data.info)

}